src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -592,6 +694,37 @@ export default class ResultsViewOncoprint extends React.Component<IResultsViewOn
             const attrIds = paramsMap[CLINICAL_TRACKS_URL_PARAM].split(",");
             attrIds.map((attrId:string)=>this.selectedClinicalAttributeIds.set(attrId, true));
         }
+        if (paramsMap[ONCOPRINT_SORTBY_URL_PARAM]) {
+            const mode = paramsMap[ONCOPRINT_SORTBY_URL_PARAM];
+            switch (mode) {
+                case SortByUrlParamValue.CASE_ID:                                         // sort by sample or patient id (a.k.a. alphabetical)
+                    this.configureSortMode('alphabetical', false, false);
+                    break;
+                case SortByUrlParamValue.CASE_LIST:                                       // sort by order of appearance in case list (when selected on query page)
+                    if (this.caseListSortPossible) {
+                        this.configureSortMode('caseList', false, false);
+                    }
+                    break;
+                case SortByUrlParamValue.DATA:                                            // sort by data
+                    this.configureSortMode('data', false, false);
+                    break;
+                case SortByUrlParamValue.DATA_MUTATION:                                        // sort by data + mutation type
+                    this.configureSortMode('data', true, false);
+                    break;
+                case SortByUrlParamValue.DATA_DRIVER:                                        // sort by  data + driver/passenger
+                    this.configureSortMode('data', false, true);
+                    break;
+                case SortByUrlParamValue.DATA_MUTATION_DRIVER:                                    // sort by data + mutation type + driver/passenger
+                    this.configureSortMode('data', true, true);
+                    break;
+            }
+        }
+    }
+
+    private configureSortMode(type:'data'|'caseList'|'alphabetical', byMutation:boolean, byDriver:boolean){
```

Comment:
should be @action

Reply:
Ah really? It is just a method that is called with real arguments in the code block above. It is not and should not be autotriggered by observable changes. Or am I wrong in this?

Reply:
action is a decorator you put on functions that change observables. you're thinking of autorun. action wraps the function in a transaction so that observable changes are not triggered until the function completes.

Reply:
please do this

Reply:
@adamabeshouse Ok, annotation was added. Sorry for the confusion.